### DISCUSSION TITLE
getting error when running as service Permission denied

### DISCUSSION THREAD

**Discussion by Pink-o** [garethgeorge/backrest#209] [Q&A] [answered]
Hi,
first of all thank you for your work.
I am planning to use backrest for organizing my backups and running it directly works without issues but as service I am getting these error:
` ERROR        task failed        {"task": onetime backup for plan \"test_doc\"", "error": "repo.Backup for repo \"my_repo\": failed to initialize repo: init failed:  command \"/home/username/.local/share/backrest/restic-0.16.4 init --json -o sftp.args=-oBatchMode=yes\" failed: exit status 1\nProcess STDOUT: \nsubprocess ssh: username@<IP>: Permission denied (publickey,password).\nFatal: create repository at sftp:username@<IP>:/home/restic/my_repo failed: unable to start the sftp session, error: error receiving version packet from server: server unexpectedly closed connection: unexpected EOF\n\n", "duration": "3.473671111s"} `

Running directly backrest and performing the same operation I get:
`	INFO	task finished	{"task": "onetime backup for plan \"test_doc\"", "duration": "4.942722071s"}`

What am I doing wrong?
I suspect backrest.service cannot access my ssh credential or something related to it?

OS: Nobara Linux 39 (GNOME Edition) x86_64 
Kernel: 6.7.6-201.fsync.fc39.x86_64

**Comment by garethgeorge** [comment]
I suspect you're right that it can't find your SSH keyfile, you may want to provide an explicit path to your SSH credentials

Take a look at this guide https://github.com/restic/restic/issues/312

You may either use flag overrides to explicitly setup the credentials, or you can edit the ssh config for the user backrest is running as.

**Reply by Pink-o** [reply]
Thank you for the reply, another bit of information: when I run the app standalone it asks me the Passphrase to unlock my ssh key and after the command proceed.
In case of service doesn't ask anything and it fails. maybe this is my problem.
Which flag can I use to set the Passphrase?

**Comment by Pink-o** [comment, marked as answer]
ok I have found the way:

- create a new key without the passphrase
- just put: ` -o sftp.args="-i /path/2/u/key"` in the flags
